tools/testrunner/junit.go

```diff
@@ -0,0 +1,207 @@
+// The MIT License
+//
+// Copyright (c) 2025 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package testrunner
+
+import (
+	"encoding/xml"
+	"fmt"
+	"iter"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/jstemmer/go-junit-report/v2/junit"
+)
+
+type junitReport struct {
+	path string
+	junit.Testsuites
+}
+
+func (j *junitReport) read() {
```

Comment:
Wondering if there's a better option to initialize the junitReport struct with an already open file instead of having the struct be mutable.

Reply:
We could probably improve it slightly, but I think we're hitting diminishing returns here now (given its just an internal tool).